### DISCUSSION TITLE
Bad default charset for `db/convert-charset`

### DISCUSSION THREAD

**Discussion by fleaz** [craftcms/cms#11823] [Ideas]
Hey,

the default charset  for the `db/convert-charset` console command currently is `utf8`, which is not optimal imho because it's just an alias for a different charset. Currently it points to `utf8mb3` (which itself is deprecated but that's another topic), but according to the MySQL docs, it will be shifted in the future to point to `utf8mb4` and therefore their [official docs](https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-sets.html) explicitly discourage one from using the alias and rather specify the actual charset you want.

Maybe that's something consider changing *before* the alias switches its value and produces unwanted results :D

**Comment by brandonkelly** [comment]
The command uses the [`charset`](https://craftcms.com/docs/4.x/config/db-settings.html#charset) database connection setting by default. All of your points are still valid though, but for the default setting value. Probably something we should change to be explicitly `utf8mb3` or `utf8mb4` in Craft 5.

**Reply by angrybrad** [reply]
Yup, I'd say `utf8mb4` for Craft 5. Looks like [5.7 supports it](https://dev.mysql.com/doc/refman/5.7/en/charset-unicode-sets.html), too.

**Comment by ItagMichael** [comment]
Just curious why this is better as a default change for Craft 5 but not Craft 4 - just too big of a change?

**Reply by brandonkelly** [reply]
@ItagMichael It will require people to run `db/convert-charset` to ensure all tables are using the same `utf8mb5` charset. Otherwise you’ll get performance issues for any queries that touch multiple tables with different charsets.
